fashion-example-query/README.md

```diff
@@ -3,83 +3,126 @@
 
-In your terminal:
+There are some requirements to run this example, to download everything you need, just run in your terminal:
 
 ```
 pip install -r requirements.txt
 ```
 
+## 🔮 Overview of the files
+
+* flows/index.yml: YAML file for indexing.
+* flows/query.yml: YAML file for querying.
+* pods/encoder.yml: YAML file for encoder pod.
+* pods/indexer-pullover.yml: Indexer for each specific category.
+* /workspace: Directory that stores indexed files (embeddings and documents). Automatically created after the first indexing.
+* Dockerfile: Dockerfile to run the example.
 
 ## 🏃 Run the Flows
 
-Now that we've got the code to load our data, we're going to dive into writing our app and running our Flows!
+We usually need at least two Flows in Jina. One for Indexing and one for Querying. 
 
 ### Index Flow
 
-To run the index you type:
+First thing we need is to index our data, and we need a Flow to manage this. 
 
-```bash
-python app.py -t index
-```
-First up we need to build up an index and then we'll search through this index when we use the query Flow later.
-So what is happening here?
-
-There are 10 categories in the fashion-mnist-data, but to simplify this example we will only use 3 categories:
+ ```python
+def index(num_doc, target: dict):
+    f = Flow.load_config('flows/index.yml')
+    with f:
+        f.index(index_generator(num_doc, target), request_size=2048)
+ ```
+      
+There are 10 categories in the fashion-mnist-data, but to simplify this example we will only use 3 of them:
 
     0	        T-shirt/top
     1	        Trouser
     2	        Pullover
-    
-We will create a separate index per category, so each category will have its own yml file and then we index the data updating updating the label
-
-```
+ 
+ We will create a separate index per category, so each category will have its own YAML file. We can then index the data updating its label:
+ 
+ ```python
 d.tags.update({'label': get_mapped_label(label_int)})
 ```
-This is where you could tweak the code if you would like to see only one category.
-For example if you would like to see only pullovers, you could do something like this
+ 
+This is where you could tweak the code if you would like to see only one category. For example if you would like to see only pullovers, you could do this:
+
+```python 
+def query_generator(num_doc: int, target: dict):
+    for j in range(num_doc):
+        n = random.randint(0, 9999) #there are 10000 query examples, so that's the limit
+        label_int = target['query-labels']['data'][n][0] 
+        category = get_mapped_label(label_int) 
+        if category == 'Pullover':
+            d = Document(content=(target['query']['data'][n]))
+            d.tags['label'] = category
+            yield d
+```
+ 
+ Then we have ready all the indexes!
 
-![alt text](.github/images/filter.png "Results")
 
-Then we have ready all the indexes!
+You can run all this with the command:
 
+```bash
+python app.py -t index
+```
 
```

Comment:
It is always good practice (especially with something as complex as Jina) to tell the user which output they should expect if anything ran correctly.